cms/djangoapps/contentstore/views/component.py

```diff
@@ -504,6 +504,12 @@ def component_handler(request, usage_key_string, handler, suffix=''):
     """
     usage_key = UsageKey.from_string(usage_key_string)
 
+
+    # Addendum:
+    # TNL 101-62 studio write permission is also checked for editing content.
+
+    if handler == 'submit_studio_edits' and not has_course_author_access(request.user, usage_key.course_key):
+        raise PermissionDenied("No studio write Permissions")
```

Comment:
It doesn't seem ideal that we only check permissions in the case of one specific handler... It seems to me like _no_ handlers should have permission to alter `Scope.content` or `Scope.settings` XBlock fields unless the user has course author access.

For example, we could instead make the call to `modulestore().update_item()` conditional on having author permission.

I guess the downside is that it's harder to throw a helpful error message if the user doesn't have permission but does call a handler that modifies settings/content-scoped fields?

Reply:
" or simply caching its values" was the line I was concerned about in gating around that call.

Reply:
I'm not really clear on what "simply caching its values" is referring to.

Reply:
>    # unintentional update to handle any side effects of handle call. could potentially be updating actual course data or simply caching its values. 
I'm really confused by what `unintentional update` means as well. In theory any call to the modulestore simply updates the document for the xblock, so it shouldn't be something a non-course-author can do, except if we have some kind of worker task which uses ajax requests which seems unlikely?
I agree I have no idea what "caching values" would do on an update-- simply just push this to the top of the cms xblock cache if called?

Reply:
I still don't really know. Does this old diff shed any light? 

![Screen Shot 2022-10-31 at 9 46 40 AM](https://user-images.githubusercontent.com/945577/199062812-eb8b1e15-4892-440b-b51e-f5cb51baf3d5.png)

Reply:
I don't know, but I'm going to make this change tomorrow try the ususal xblock operations (edit, publish, etc) on a bunch of different blocks and see if it breaks anything. I agree that putting the gating here might be more reasonable, and worth doing. As is the case with these things, this might be a "try it see what breaks" senario.

Reply:
https://github.com/openedx/edx-platform/pull/31239 I made the change against master, as I think that makes more sense.  Let me know what you think.